html/js/scrollNav.js

```diff
@@ -0,0 +1,141 @@
+        let position = window.scrollY + 200
+        navbarlinks.forEach(navbarlink => {
+            if (!navbarlink.hash) return
+            let section = select(navbarlink.hash)
+            if (!section) return
+            if (position >= section.offsetTop && position <= (section.offsetTop + section.offsetHeight)) {
+                navbarlink.classList.add('active')
+            } else {
+                navbarlink.classList.remove('active')
+            }
+        })
+    }
+    window.addEventListener('load', navbarlinksActive)
+    onscroll(document, navbarlinksActive)
+
+    /**
+     * Scrolls to an element with header offset
+     */
+    const scrollto = (el) => {
+        let header = select('#upNav')
+        let offset = header.offsetHeight
+
+        if (!header.classList.contains('header-scrolled')) {
+            offset -= 10
+        }
+
+        let elementPos = select(el).offsetTop
+        window.scrollTo({
+            top: elementPos - offset,
+            behavior: 'smooth'
+        })
+    }
+
+    
+    /**
+     * Back to top button
+     */
+    let backtotop = select('.back-to-top')
+    if (backtotop) {
+        const toggleBacktotop = () => {
+            if (window.scrollY > 100) {
+                backtotop.classList.add('active')
+            } else {
+                backtotop.classList.remove('active')
+            }
+        }
+        window.addEventListener('load', toggleBacktotop)
+        onscroll(document, toggleBacktotop)
+    }
+
+    /**
+     * Mobile nav toggle
+     */
+    // on('click', '.mobile-nav-toggle', function(e) {
+    //     select('#navbar').classList.toggle('navbar-mobile')
+    //     this.classList.toggle('bi-list')
+    //     this.classList.toggle('bi-x')
+    // })
+
+    /**
+     * Mobile nav dropdowns activate
+     */
+    //on('click', '.navbar .dropdown > a', function(e) {
+    //    if (select('#navbar').classList.contains('navbar-mobile')) {
+    //        e.preventDefault()
+    //        this.nextElementSibling.classList.toggle('dropdown-active')
+    //    }
+    //}, true)
+
+    /**
+     * Scrool with ofset on links with a class name .scrollto
+     */
+    on('click', '.scrollto', function(e) {
+        if (select(this.hash)) {
+            e.preventDefault()
+
+            let navbar = select('#navbar')
+            if (navbar.classList.contains('navbar-mobile')) {
+                navbar.classList.remove('navbar-mobile')
+                let navbarToggle = select('.mobile-nav-toggle')
+                navbarToggle.classList.toggle('bi-list')
+                navbarToggle.classList.toggle('bi-x')
+            }
+            scrollto(this.hash)
+        }
+    }, true)
+
+    /**
+     * Scroll with ofset on page load with hash links in the url
+     */
+    window.addEventListener('load', () => {
+        if (window.location.hash) {
+            if (select(window.location.hash)) {
+                scrollto(window.location.hash)
```

Comment:
```suggestion
                setTimeout(scrollto, 10, window.location.hash)
```

may resolve https://github.com/openfoodfacts/openfoodfacts-server/issues/7415